compiler/src/iree/compiler/Codegen/Common/CPU/CPULowerToUKernels.cpp

```diff
@@ -244,8 +239,15 @@ matchDAGForUKernel(RewriterBase &rewriter, linalg::Mmt4DOp op,
   Value flagsVal = rewriter.create<arith::ConstantOp>(
       loc, rewriter.getI32IntegerAttr(flags));
   auto fn = getFnNameAndDefAttrs(ukernelName, rewriter, targetAttr);
+  SmallVector<Type> returnTypes{outType};
+  if (!isVMVXBackend(targetAttr)) {
+    // Hack to avoid issues with void-returning functions in llvm-cpu.
```

Comment:
Not sure I would characterize this as a hack. This is a legit thing to do if the ukernel is returning a scalar value.

Reply:
but it's not! it's really a void-returning kernel and we're making it return an int... as a hack.